
syntax = "proto3";

package tc_serialization;


message Coordinates{
 	double lat = 1;
	double lng = 2;
}


message Stop{
 	string name = 1;
	Coordinates place = 2;
	uint32 id = 3; 
}

message Bus{
 	string name = 1;
	repeated Stop track = 2;
	bool is_round = 3;
	uint32 id = 4;
}


message Distance{
 	string stop1 = 1;
 	string stop2 = 2;
	uint32 dist = 3;
}

message RouteInfo{
        double bus_wait_time = 1;
        double bus_velocity = 2;
        double bus_leaving_time = 3;
}

message ColorVariant{
	uint32 type = 1;
 	string color =2;
        uint32 red = 3;
        uint32 green = 4;
        uint32 blue = 5;
        double opacity = 6;
}

message MapRenderSettings {
        double width = 1;
        double height = 2;
        double padding = 3;
        double line_with = 4;
        double stop_radius = 5;

        uint32 bus_label_font_size = 6;
        double bus_label_offset_first = 7;
        double bus_label_offset_second = 8;

        uint32 stop_label_font_size = 9;
        double stop_label_offset_first = 10;
        double stop_label_offset_second = 11;

        ColorVariant underlayer_color = 12;

        double underlayer_width = 13;
        repeated ColorVariant color_palette = 14;

}

message Edge_Route {
        uint32 from = 1;
        uint32 to = 2;
        double weight = 3;
        uint32 transport_id = 4;
        uint32 span_count = 5;
}



message TransportCatalogue{
	repeated Stop stops = 1;
	repeated Bus buses = 2;
	repeated Distance dist_list= 3;
	RouteInfo route_info = 4;
	MapRenderSettings map_settings = 5;
	repeated Edge_Route all_graphs = 6;
}

